### PR TITLE
VHM Content

### PR DESCRIPTION
**Pull request by Loquacity**
Edit Virt section in Client Config, move k8s from Admin to Client Config, add new section for VHM that is de-VMwared. Also adds new sections for Azure, AWS, GCE to be populated.

https://github.com/SUSE/spacewalk/issues/9501

### CODE REVIEW

**Comment by Loquacity** [comment]
image


image

**Comment by Loquacity** [comment]
That's probably enough for one PR ;)

File: modules/client-configuration/pages/vhm.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
+[[virt-vhm]]
+= Virtual Host Managers
+
+Virtual Host Managers (VHMs) are used to gather information from a range of client types.
+
+You can use a VHM to manage public cloud instances that are not directly supported by {productname}.
```
Comment (meaksh) [inline modules/client-configuration/pages/vhm.adoc]: IMHO it would be better to rephrase this to something like:

_You can use a VHM to collect your Private and Public Cloud instances and expose them into SUSE Manager_

I think it's not correct to say "manage public cloud instances that are not directly **supported** by SUSE Manager"

Comment (Loquacity) [reply modules/client-configuration/pages/vhm.adoc]: Those sentences are saying different things, though:

The first sentence says "VHMs can manage VM clients".
The second sentence says "VHMs can manage VM clients that aren't managed in other ways".

I agree that use of the word "supported" is confusing, though. Basically, we have Xen and KVM that are managed directly, and we have VHMs which are used for managing everything else. How would you word that?

Comment (meaksh) [reply modules/client-configuration/pages/vhm.adoc]: The thing is that with VHM you cannot really manage the collected virtual instances. You only see them as virtualization instances.

Only in case that some of those virtual instances are actually registered as clients in SUMA (trad or minion), then you would be able to manage them.

In an ideal scenario, you would have all your virtual instances already registered in SUMA, then after adding the VHM that host those instances, you would get the proper virtualization groups and will see those instances now as belonging to this particular VHM. This is really important for a proper subscription matching (specially for 1-2 virtual machine subscription).

@Loquacity in case you want to clarify this better, we can maybe have a meeting in mumble. Just ping me :smile: :+1:

Comment (Loquacity) [reply modules/client-configuration/pages/vhm.adoc]: OK, I've changed "managed" to "communicate" in this instance. I think a little more exposition might be required, though.

**Comment by Loquacity** [comment]
@meaksh I've added everything you gave me now (I think!), can you please take another look?

File: modules/client-configuration/pages/vhm.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+Regions are then further divided into zones.
+For example, the [systemitem]``US-East`` region might contain zones called [systemitem]``us-east-2a`` and [systemitem]``us-east-2b``, among others.
+
+{suse} uses the zone of a virtual machine to determine the appropriate subscription to provide.
+If all of your VMs are provided by the same zone, you are within the terms and conditions of the ``1-2 Virtual Machines`` subscription.
+
+If your VMs are provided by different zones, even if they are within the same region, you might not meet the conditions of the ``1-2 Virtual Machines`` subscription.
+In this case, check your subscription carefully.
+
+For more information, https://www.suse.com/products/terms_and_conditions.pdf or contact {suse}.
```
Comment (keichwa) [inline modules/client-configuration/pages/vhm.adoc]:
```suggestion
For more information, see https://www.suse.com/products/terms_and_conditions.pdf or contact {suse}.
```

File: modules/client-configuration/pages/vhm-aws.adoc [hunk 3]
```diff
@@ -0,0 +1,43 @@
+[[vhm-aws]]
+= VHM and Amazon Web Services
+
+You can use a {productname} VHM to manage Amazon Web Services (AWS) clusters.
```
Comment (meaksh) [inline modules/client-configuration/pages/vhm-aws.adoc]:
```suggestion
You can use a {productname} VHM to gather your instances from Amazon Web Services (AWS).
```

File: modules/client-configuration/pages/vhm-aws.adoc [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+The VHM allows {productname} to obtain and report information about your clusters, including the ability to do CVE audits, and report if containers are outdated.
```
Comment (meaksh) [inline modules/client-configuration/pages/vhm-aws.adoc]: Why are we mentioning containers here? In any case, using a VHM does not enable you to do CVE audits or getting info about patches. You must have those instances also registered as systems into SUSE Manager in order to manage them.

File: modules/client-configuration/pages/vhm-aws.adoc [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+For more information on VHMs, see xref:client-configuration:vhm.adoc[].
+
+
+
+== Create an Amazon EC2 VHM
+
+
+The Virtual Host Manager (VHM) runs on the {productname} Server.
+
+To run a VHM, your {productname} Server will need to have port 443 open, in order to access the clients.
```
Comment (meaksh) [inline modules/client-configuration/pages/vhm-aws.adoc]: The `443` port is not a requirement here.

File: modules/client-configuration/pages/vhm-azure.adoc [hunk 6]
```diff
@@ -0,0 +1,44 @@
+[[vhm-azure]]
+= VHM and Azure
+
+You can use a {productname} VHM to manage Microsoft Azure clusters.
```
Comment (meaksh) [inline modules/client-configuration/pages/vhm-azure.adoc]: Same than for AWS. VHM does not enable you to manage your instances/clusters.

File: modules/client-configuration/pages/vhm-azure.adoc [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+The VHM allows {productname} to obtain and report information about your clusters, including the ability to do CVE audits, and report if containers are outdated.
```
Comment (meaksh) [inline modules/client-configuration/pages/vhm-azure.adoc]: Same than AWS

File: modules/client-configuration/pages/vhm-azure.adoc [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+For more information on VHMs, see xref:client-configuration:vhm.adoc[].
+
+
+
+== Create an Azure VHM
+
+
+The Virtual Host Manager (VHM) runs on the {productname} Server.
+
+To run a VHM, your {productname} Server will need to have port 443 open, in order to access the clients.
```
Comment (meaksh) [inline modules/client-configuration/pages/vhm-azure.adoc]: Same as AWS

File: modules/client-configuration/pages/vhm-caasp.adoc [hunk 9]
```diff
@@ -0,0 +1,7 @@
+[[kubernetes]]
+= VHM and SUSE Container-as-a-Service Platform
+
+You can use a {productname} VHM to manage SUSE Container-as-a-Service Platform ({caasp}) clusters.
```
Comment (meaksh) [inline modules/client-configuration/pages/vhm-caasp.adoc]: Same

File: modules/client-configuration/pages/vhm-gce.adoc [hunk 10]
```diff
@@ -0,0 +1,47 @@
+[[vhm-hce]]
+= VHM and Google Compute Engine
+
+You can use a {productname} VHM to manage Google Compute Engine (GCE) clusters.
```
Comment (meaksh) [inline modules/client-configuration/pages/vhm-gce.adoc]: Again, the same :)

File: modules/client-configuration/pages/vhm-gce.adoc [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+The VHM allows {productname} to obtain and report information about your clusters, including the ability to do CVE audits, and report if containers are outdated.
```
Comment (meaksh) [inline modules/client-configuration/pages/vhm-gce.adoc]: Same

File: modules/client-configuration/pages/vhm-gce.adoc [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+For more information on VHMs, see xref:client-configuration:vhm.adoc[].
+
+
+
+== Create a GCE VHM
+
+
+The Virtual Host Manager (VHM) runs on the {productname} Server.
+
+To run a VHM, your {productname} Server will need to have port 443 open, in order to access the clients.
```
Comment (meaksh) [inline modules/client-configuration/pages/vhm-gce.adoc]: Same

File: modules/client-configuration/pages/vhm.adoc [hunk 13]
```diff
@@ -0,0 +1,35 @@
+[[virt-vhm]]
+= Virtual Host Managers
+
+Virtual Host Managers (VHMs) are used to gather information from a range of client types.
+
+VHMs can be used to collect public cloud instances and organize them into virtualization groups.
```
Comment (meaksh) [inline modules/client-configuration/pages/vhm.adoc]:
```suggestion
VHMs can be used to collect private or public cloud instances and organize them into virtualization groups.
```

File: modules/reference/pages/systems/virtual-host-managers.adoc [hunk 14]
```diff
@@ -1,36 +1,16 @@
 [[ref.webui.systems.virt-host-managers]]
 = Virtual Host Managers
 
-Third party hypervisors and hypervisor managers are called Virtual Host Managers (VHMs).
-VHMs can manage one or more virtual hosts, and each virtual host can contain one or more virtual guests.
-VHMs can be based on VMWare, or created from a file.
+Virtual Host Managers (VHMs) are used to gather information from a range of client types.
 
+VHMs can be used to collect public cloud instances and organize them into virtualization groups.
```
Comment (meaksh) [inline modules/reference/pages/systems/virtual-host-managers.adoc]:
```suggestion
VHMs can be used to collect private or public cloud instances and organize them into virtualization groups.
```

File: modules/client-configuration/nav-client-config-guide.adoc [hunk 15]
```diff
@@ -104,6 +111,19 @@ include::modules/client-configuration/pages/virt-vmware.adoc[leveloffset=+2]
 
 include::modules/client-configuration/pages/virt-file.adoc[leveloffset=+2]
 
+// VHMs
+include::modules/client-configuration/pages/vhm.adoc[leveloffset=+1]
+
+include::modules/administration/pages/vhm-azure.adoc[leveloffset=+2]
```
Comment (paususe) [inline modules/client-configuration/nav-client-config-guide.adoc]: Link is wrong, this page is under modules/client-configuration/pages/vhm-azure.adoc
Is that the right place?

File: modules/client-configuration/nav-client-config-guide.adoc [hunk 16]
```diff
@@ -104,6 +111,19 @@ include::modules/client-configuration/pages/virt-vmware.adoc[leveloffset=+2]
...
+
+include::modules/administration/pages/vhm-aws.adoc[leveloffset=+2]
```
Comment (paususe) [inline modules/client-configuration/nav-client-config-guide.adoc]: Same: this page lives under modules/client-configuration/pages/vhm-aws.adoc

File: modules/client-configuration/nav-client-config-guide.adoc [hunk 17]
```diff
@@ -104,6 +111,19 @@ include::modules/client-configuration/pages/virt-vmware.adoc[leveloffset=+2]
...
+
+include::modules/administration/pages/vhm-gce.adoc[leveloffset=+2]
```
Comment (paususe) [inline modules/client-configuration/nav-client-config-guide.adoc]: Same: actual path is modules/client-configuration/pages/vhm-gce.adoc

**Comment by Loquacity** [comment]
> Some wrong links, didn't read the actual content

Oof, good spot! Fixed now :)

**Comment by Loquacity** [comment]
Created the mother of all merge conflicts here. Recreated in https://github.com/SUSE/doc-susemanager/pull/862
